### DISCUSSION TITLE
How to get Authenticated User in controller?

### DISCUSSION THREAD

**Discussion by Martijn-Bos** [lexik/LexikJWTAuthenticationBundle#1071] [Q&A]
I'm sending an incoming request on an API endpoint with valid jwt token. I'd like to get the attached User class in the controller or later in the process. How to do this?

How do I here get the user entity connected to that token?

**Comment by fd6130** [comment]
In your controller, you can use `$this->getUser()`.

Or you can inject security service like the doc said so: https://symfony.com/doc/current/security.html#fetching-the-user-from-a-service

**Reply by Martijn-Bos** [reply]
`$this->getUser() `in controller returns an empty array. 

```
class ParentCollectionController extends AbstractController
{
    private $security;

    public function __construct(Security $security)
    {
        // Avoid calling getUser() in the constructor: auth may not
        // be complete yet. Instead, store the entire Security object.
        $this->security = $security;
    }

    #[Route('/api/collection', name: 'app_parent_collection')]
    public function index(): Response
    {

        return $this->json($this->security->getToken());
    }
```

Either `$this->security->getUser()` or `$this->security->getToken()` returns an empty array.

**Reply by fd6130** [reply]
Seems like your route is not under authentication, more like a `public_access` to me.

Can you check your `security.yaml` to make sure the route you access is under authenticate protection?
